src/UUIDType.php

```diff
@@ -0,0 +1,51 @@
+<?php declare(strict_types=1);
+
+namespace Fortuneglobe\Types;
+
+use Fortuneglobe\Types\Exceptions\InvalidUUIDValueException;
+use Fortuneglobe\Types\Interfaces\RepresentsUUIDValue;
+
+/**
+ * Class UUIDType
+ * @package Fortuneglobe\Types
+ */
+class UUIDType extends StringType implements RepresentsUUIDValue
+{
+	private const NIL          = '00000000-0000-0000-0000-000000000000';
+
+	private const UUID_PATTERN = '^[0-9A-Fa-f]{8}-[0-9A-Fa-f]{4}-[0-9A-Fa-f]{4}-[0-9A-Fa-f]{4}-[0-9A-Fa-f]{12}$';
```

Comment:
I would use the case insensitive switch and shorten the pattern.

Reply:
Good catch. Fixed in a8b71ebb9187a6b6f7dd36c139fbc47be3830e1e